### DISCUSSION TITLE
TIF-to-JPG and NEF-to-JPG conversion issues

### DISCUSSION THREAD

**Discussion by SailorGuy57** [ImageMagick/ImageMagick#4165] [Help]
[Magick_Convert2JPG.pdf](https://github.com/ImageMagick/ImageMagick/files/7124406/Magick_Convert2JPG.pdf)

< See the attached PDF for details and images >

I am a new user of ImageMagick and am unfamiliar with all the conversion options.

I have a requirement to batch convert and rename multiple PNG, NEF, and TIF files to JPG format.

I don’t need to adjust any of the settings or properties of the resulting JPG file – just a simple convert “as is” from PNG / NEF / TIF to JPG.

I am using a very simple command string:

•	Magick   -verbose  image.png   image.jpg
•	Magick   -verbose  image.nef   image.jpg
•	Magick   -verbose  image.tif     image.tif

The PNG images are converting fine.

However, when the NEF (proprietary Nikon TIF format) images convert, the resulting JPG files are lighter / brighter than the original NEF file.

And when the TIF images convert, I get a warning message and instead of a single resulting JPG file, I get two files – image-0.jpg (at the same source dimensions) and image-1.jpg (at a significantly reduced thumbnail size).

See the following pages in the attached PDF for:

•	The “before” and “after” images
•	The console output of the “magick” commands
•	The EXIF data for the “before” and “after” images

Any guidance for this newbie will be greatly appreciated!

Regards,

Sailor Guy

**Comment by snibgo** [comment]
Your PDF shows that your TIFF contains two images, sized 2560x1920 and 160x120 pixels. That is why you get two output jpegs. If you want only the first, then:
```
magick -verbose in.tif[0] out.jpg
```
The warning is about a "GainControl" tag. It hasn't caused any problems.

The NEF file contains a raw image. "Raw" means each pixel represents only one colour, and other characteristics. IM converts this to an image with RGB channels. There are some options you can use. See https://imagemagick.org/script/defines.php , the defines that start with "dng:". Specify these _before_ reading the NEF file. You might want "-define dng:no-auto-bright=false" if you don't want the auto-brightening, and maybe "-define dng:use-camera-wb=true" to use the camera's setting for white balance.

Beware that your claims about the lightness of "the original NEF file" are not accurate. You haven't shown any original NEF file. What you have showed is a result that has been processed from a NEF file. So you are comparing two different processes (ImageMagick versus some other process), not IM versus some "original".

**Reply by zdfvip153** [reply]
How do I print the last frame

**Comment by urban-warrior** [comment]
Use this command:

```
magick 'in.tif[-1]' in.jpg
```

However, it requires a patch @ https://github.com/ImageMagick/ImageMagick/commit/53eb353a1a4487c5dcf887e11fc7381f2deb08f0 to work properly.  Other images formats work as expected for retrieving the last frame.

**Reply by zdfvip153** [reply]
请问这种有阿拉法通道的tif怎么转透明背景的png
May I ask how to convert this tif with alpha channel to png with transparent background
[希音.zip](https://github.com/ImageMagick/ImageMagick/files/9757103/default.zip)

**Comment by snibgo** [comment]
Your zip file contains two TIFF files. Each of those contains multiple images. Each image is encoded as CMYK. Do you want just the first image in each file, or all images? I assume you want just the first.
```
f:\web\im>%IMG7%magick "1 (1).tif" info:
1 (1).tif[0] TIFF 3085x3268 3085x3268+0+0 8-bit ColorSeparation CMYK 59600000B 0
.016u 0:00.007
1 (1).tif[1] TIFF 3085x3268 3085x3268+0+0 8-bit CMYK 59600000B 0.016u 0:00.013
1 (1).tif[2] TIFF 3084x3267 3084x3267+0+0 8-bit CMYK 59600000B 0.016u 0:00.018
1 (1).tif[3] TIFF 3084x2741 3084x2741+0+526 8-bit CMYK 59600000B 0.016u 0:00.022
1 (1).tif[4] TIFF 2595x3211 2595x3211+288+0 8-bit CMYK 59600000B 0.016u 0:00.029
1 (1).tif[5] TIFF 3084x3267 3084x3267+0+0 8-bit CMYK 59600000B 0.016u 0:00.034
1 (1).tif[6] TIFF 1182x743 1182x743+1028+0 8-bit CMYK 59600000B 0.016u 0:00.040
1 (1).tif[7] TIFF 1667x589 1667x589+396+998 8-bit CMYK 59600000B 0.016u 0:00.046
1 (1).tif[8] TIFF 414x428 414x428+288+1988 8-bit CMYK 59600000B 0.000u 0:00.063
1 (1).tif[9] TIFF 1322x710 1322x710+1172+1951 8-bit CMYK 59600000B 0.000u 0:00.092
1 (1).tif[10] TIFF 1682x428 1682x428+1201+2783 8-bit CMYK 59600000B 0.000u 0:00.118

f:\web\im>%IMG7%magick "1 (2).tif" info:
1 (2).tif[0] TIFF 2439x3365 2439x3365+0+0 8-bit ColorSeparation CMYK 51.1039MiB
0.016u 0:00.012
1 (2).tif[1] TIFF 2439x3365 2439x3365+0+0 8-bit CMYK 51.1039MiB 0.016u 0:00.029
1 (2).tif[2] TIFF 2439x3365 2439x3365+0+0 8-bit CMYK 51.1039MiB 0.016u 0:00.045
1 (2).tif[3] TIFF 2439x3365 2439x3365+0+0 8-bit CMYK 51.1039MiB 0.016u 0:00.060
1 (2).tif[4] TIFF 2439x3365 2439x3365+0+0 8-bit CMYK 51.1039MiB 0.016u 0:00.082
1 (2).tif[5] TIFF 2439x3365 2439x3365+0+0 8-bit CMYK 51.1039MiB 0.000u 0:00.104
1 (2).tif[6] TIFF 2209x858 2209x858+126+181 8-bit CMYK 51.1039MiB 0.000u 0:00.125
1 (2).tif[7] TIFF 1123x778 1123x778+739+1119 8-bit CMYK 51.1039MiB 0.000u 0:00.151
```
PNG files cannot record CMYK, so the images must be converted to sRGB (or other RGB colorpace).

Looking in more detail at one of the images:
```
magick "1 (1).tif[0]" -verbose info:

  Profiles:
    Profile-8bim: 286378 bytes
    Profile-icc: 557168 bytes
  :
  :
  Properties:
  :
  :
    icc:copyright: Copyright 2002 Adobe Systems, Inc.
    icc:description: Japan Color 2001 Coated
    photoshop:ColorMode: 4
    photoshop:ICCProfile: Japan Color 2001 Coated
```
So that image has an embedded ICC colour profile. To convert it to sRGB, we shold use "-profile".
```
magick "1 (1).tif[0]" -profile sRGB.icc out.png
```
But perhaps in your input, the alpha channel represents transparency instead of opacity. If so, then you can negate it:
```
magick "1 (1).tif[0]" -profile sRGB.icc -channel A -negate +channel out2.png
```

**Reply by zdfvip153** [reply]
Thank you very much.

**Reply by zdfvip153** [reply]
If you choose to save the transparency in ps, the converted png will be white. How to determine whether the transparency is saved to choose different conversion methods

**Comment by snibgo** [comment]
I don't use Photoshop.

I have heard that when Photoshop saves images, it writes an extra image at [0], and that image is a flattened version of all the other images. "Flattened" means it has no transparency.

So you should check with "`magick in.ext info:`" how many images there are. You may need to process image [1] instead of [0].

**Reply by fmw42** [reply]
In a PSD file, usually the first layer that IM gets is the flattened layer (from all the layers).  But that is not usually the case with TIFF.

**Reply by snibgo** [reply]
Ah, thanks.
